src/logger.ts

```diff
@@ -0,0 +1,27 @@
+import { LogFunctionFactory, LogLevel, LogMeta } from "@graphile/logger";
+
+export { LogLevel as MigrateLogLevel } from "@graphile/logger";
+export type MigrateLogFactory = LogFunctionFactory<Record<string, unknown>>;
+export interface MigrateLogMeta extends LogMeta {
+  error?: Error;
+}
+
+export const migrateLogFactory: MigrateLogFactory = () => {
+  return (level: LogLevel, message: string, meta?: MigrateLogMeta): void => {
+    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+    const method = (() => {
+      switch (level) {
+        case LogLevel.ERROR:
+          return "error" as const;
+        case LogLevel.WARNING:
+          return "warn" as const;
+        case LogLevel.INFO:
+        default:
+          return "log" as const;
+      }
+    })();
+
+    // eslint-disable-next-line no-console
+    meta ? console[method](message, meta) : console[method](message);
```

Comment:
I think we should resist logging the meta object and instead put the relevant content into the message. This means that we can add more information into meta (e.g. durations, etc) over time without cluttering up the printed output. You'll notice I already added the error message above for this reason.

Reply:
If we can use the built in console log factory that'd be great, but if it messes up the layout then it's okay to use our own here.

Reply:
Managed to use the default factory with some adjustments to the format and format parameters, otherwise, logs would look like this:
```text
INFO: graphile-migrate: Already up to date ({})
INFO: [2021-04-21T05:57:55.399Z]: Running current.sql ({})
INFO: [2021-04-21T05:57:55.410Z]: current.sql unchanged, skipping migration ({})
INFO: [2021-04-21T05:57:55.420Z]: Finished (21ms) ({})
```

Instead of this:
```text
graphile-migrate: Already up to date
[2021-04-21T05:12:43.690Z]: Running current.sql
[2021-04-21T05:12:43.708Z]: current.sql unchanged, skipping migration
[2021-04-21T05:12:43.714Z]: Finished (24ms)
```